controller/string_cache.go

```diff
@@ -0,0 +1,28 @@
+package controller
+
+import (
+	"time"
+
+	gocache "github.com/patrickmn/go-cache"
```

Comment:
@xh3b4sd I found a library which quite fits. I think the performance will be better and we don't have this `1000` magic which is difficult to explain. PTALA.